src/main/java/org/jboss/xavier/sanitizer/XavierPayloadSanitizer.java

```diff
@@ -0,0 +1,169 @@
+                continue;
+            }else if (issue.startsWith("###"))
+            {
+                currentSectionWanted = false;
+                continue;
+            }
+            if(currentSectionWanted)
+            {
+                returnIssues.add(issue);
+            }
+        }
+
+        return returnIssues;
+
+    }
+
+    private static String sanitize(String fileToBeSanitized, List<String> vmsIssuesConditions, List<String> hostsIssuesConditions, List<String> clustersIssuesConditions)
+    {
+        String vmsInvalidElementIds = findInvalidElements(fileToBeSanitized, vmsIssuesConditions);
+        String hostsInvalidElementIds = findInvalidElements(fileToBeSanitized, hostsIssuesConditions);
+        String clustersInvalidElementIds = findInvalidElements(fileToBeSanitized, clustersIssuesConditions);
+
+        String resultString = retireVms(fileToBeSanitized,"$.ManageIQ::Providers::Vmware::InfraManager[*].vms[?(@.id in [" + vmsInvalidElementIds+ "])]");
+        resultString = retireVms(resultString,"$.ManageIQ::Providers::Vmware::InfraManager[*].vms[?(@.host.ems_ref in [" + hostsInvalidElementIds+ "])]");
+
+
+
+        return resultString;
+    }
+
+
+
+    private static String findInvalidElements(String fileToBeSanitized, List<String> issuesConditions)
+    {
+        Configuration conf = Configuration.builder().options(Option.ALWAYS_RETURN_LIST, Option.SUPPRESS_EXCEPTIONS).build();
+        DocumentContext parsedFile = JsonPath.using(conf).parse(fileToBeSanitized);
+
+        List<Object> errorIds = new LinkedList<>();
+
+        issuesConditions.forEach(issuesCondition -> {
+            errorIds.addAll(parsedFile.read(issuesCondition));
+        });
+
+        List<String> errorIdStringsList = new ArrayList<>();
+
+        errorIds.forEach(id -> errorIdStringsList.add(String.valueOf(id)));
+
+        return String.join(",",errorIdStringsList);
+    }
+
+    private static String retireVms(String fileToBeSanitized, String vmSelectionQuery)
+    {
+        Configuration conf = Configuration.builder().options(Option.AS_PATH_LIST, Option.ALWAYS_RETURN_LIST, Option.SUPPRESS_EXCEPTIONS).build();
+        DocumentContext parsedFile = JsonPath.using(conf).parse(fileToBeSanitized);
+
+        List<String> errorPaths = new ArrayList<String>();
+
+        errorPaths.addAll(parsedFile.read(vmSelectionQuery));
+
+
+        errorPaths.stream().forEach( errorPath -> parsedFile.put(errorPath,"retired",new Integer(1)));
+
+        return parsedFile.jsonString();
+    }
+
+    private static void writeStringToFile(String path, String fileText)
+    {
+        if (path == null || path.isEmpty())
+        {
+            String[] splitPath = inputFileName.split("\\.");
+            path = splitPath[0].concat("_sanitized.json");
+        }
+
+        try (Writer writer = new BufferedWriter(new OutputStreamWriter(
+                new FileOutputStream(path), "utf-8"))) {
+            writer.write(fileText);
+        }
```

Comment:
I've verified that if we have a preexisting `myfilename_santized.json` file, then it will be overridden after the execution of this code. Is there a way we can avoid overriding a file without notifying the user before? Here are some ideas:
- Instead of just `_sanitized.json` we can add `_satinized_date.json`
- Ask the user if he wants to override the preexisted fle.
- Just print and notify the user that is the previous file has been overridden.

Reply:
Added the date